src/Console/Command/MigrateCommand.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace MX\PhinxMigrations\Console\Command;
+
+use Magento\Setup\Console\Command\DbStatusCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Process\Process;
+
+/**
+ * @package MX\PhinxMigrations
+ * @author  James Halsall <james.halsall@inviqa.com>
+ */
+class MigrateCommand extends Command
+{
+    protected function configure()
+    {
+        $this
+            ->setName('mx:migrate')
+            ->setDescription('Migrates the database so it is ready for use with the current application code.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $dbStatusCommand = $this->getApplication()->get('setup:db:status');
+
+        $output->writeln('<info>Checking if setup:upgrade is required...</info>');
+        $exit = $dbStatusCommand->run(new ArrayInput([]), new NullOutput());
+
+        if ($exit === DbStatusCommand::EXIT_CODE_UPGRADE_REQUIRED) {
+            $output->writeln('<info>A setup:upgrade is required, running...</info>');
+            $setupUpgradeCommand = $this->getApplication()->get('setup:upgrade');
+            $setupUpgradeCommand->run(new ArrayInput([]), $output);
+        } else {
+            $output->writeln('<info>No setup:upgrade required.</info>');
+        }
+
+        $phinx = new Process('bin/phinx migrate');
```

Comment:
maybe use `vendor/robmorgan/phinx/bin/phinx migrate` here in case `"bin-dir": "bin"` is not specified in the project json (so it won't appear in the bin folder) ?